tensorflow/compiler/tf2tensorrt/trt_convert_test.cc

```diff
@@ -0,0 +1,210 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#if GOOGLE_CUDA && GOOGLE_TENSORRT
+
+#include "trt_convert.h"
+
+#include "tensorflow/cc/ops/standard_ops.h"
+#include "tensorflow/compiler/tf2tensorrt/trt_convert.h"
+#include "tensorflow/core/framework/tensor_testutil.h"
+#include "tensorflow/core/lib/core/status_test_util.h"
+#include "tensorflow/core/platform/test.h"
+#include "tensorflow/core/protobuf/meta_graph.pb.h"
+#include "tensorflow/core/public/session.h"
+
+namespace tensorflow {
+namespace tensorrt {
+
+struct TestParam {
+  TfTrtConversionParams conv_params;
+  std::vector<std::vector<int64>> input_shapes;
+};
+
+class TrtConverterTest : public ::testing::TestWithParam<TestParam> {
+ protected:
+  TrtConverterTest() { param_ = GetParam(); }
+
+  // Returns the following graph: output = input * [42, 137] + input
+  MetaGraphDef GetModel() {
+    // std::vector<int64> dims{-1, 2};
+    PartialTensorShape shape({-1, 2});
+    // TF_CHECK_OK(TensorShapeUtils::MakeShape(dims, &shape));
+
+    Scope root = Scope::NewRootScope();
+    auto c = ops::Const(root.WithOpName("my_const"), {{42.0f, 137.0f}});
+    const auto attrs = ops::Placeholder::Shape(shape);
+    auto x = ops::Placeholder(root.WithOpName("input"), DT_FLOAT, attrs);
+    auto y = ops::Mul(root.WithOpName("my_mul"), x, c);
+    auto z = ops::Add(root.WithOpName("my_add"), x, y);
+    auto q = ops::Identity(root.WithOpName("output"), z);
+
+    MetaGraphDef out;
+    root.ToGraphDef(out.mutable_graph_def());
+
+    TensorShapeProto shape_proto;
+    shape.AsProto(&shape_proto);
+    SignatureDef signature_def;
+    (*signature_def.mutable_inputs())["input"].set_name("input:0");
+    (*signature_def.mutable_inputs())["input"].set_dtype(DT_FLOAT);
+    (*signature_def.mutable_inputs())["input"].mutable_tensor_shape()->CopyFrom(
+        shape_proto);
+    (*signature_def.mutable_outputs())["output"].set_name("output:0");
+    (*signature_def.mutable_outputs())["output"].set_dtype(DT_FLOAT);
+    (*signature_def.mutable_outputs())["output"]
+        .mutable_tensor_shape()
+        ->CopyFrom(shape_proto);
+    (*out.mutable_signature_def())["serving_default"] = signature_def;
+
+    VLOG(2) << signature_def.DebugString();
+    return out;
+  }
+
+  void CheckTrtNode(const GraphDef& converted_graph_def) {
+    // Confirm that we have a TRT node with the correct attributes
```

Comment:
Shall we move put this as a comment for the function?
Add a period to the end of the line, to be consistent with the rest.